src/main/java/com/iexec/core/replicate/ReplicatesService.java

```diff
@@ -404,26 +491,29 @@ private boolean verifyStatus(String chainTaskId, String walletAddress, Replicate
         }
     }
 
-    private boolean updateReplicateWeight(String chainTaskId, Replicate replicate) {
+    private boolean isWorkerWeightCorrect(String chainTaskId,
+                                          Replicate replicate,
+                                          int workerWeight) {
         String walletAddress = replicate.getWalletAddress();
-        int workerWeight = iexecHubService.getWorkerWeight(walletAddress);
         if (workerWeight == 0) {
             log.error("Failed to get worker weight [chainTaskId:{}, workerWallet:{}]",
                     chainTaskId, walletAddress);
             return false;
         }
+        return true;
+    }
 
-        Optional<ChainContribution> chainContribution =
-                iexecHubService.getChainContribution(chainTaskId, walletAddress);
+    private boolean isChainContributionCorrect(String chainTaskId,
```

Comment:
chainContributionResultHash should not be empty as well

Reply:
Fixed, thanks!